packages/cpp/ArmoniK.Api.Common/header/objects/Task.h

```diff
@@ -0,0 +1,64 @@
+#pragma once
+
+#include "tasks_common.pb.h"
+#include <string>
+#include <vector>
+
+namespace armonik {
+namespace api {
+namespace common {
+struct TaskCreation {
+  /**
+   * Payload Id
+   */
+  std::string payload_id;
+  /**
+   * Expected output keys
+   */
+  std::vector<std::string> expected_output_keys;
+
+  /**
+   * Data dependencies, none by default
+   */
+  std::vector<std::string> data_dependencies = {};
+
+  /**
+   * Per task task options, none by default
+   */
+  armonik::api::grpc::v1::TaskOptions taskOptions = get_no_task_options();
+
+  /**
+   * Default "no task option" value
+   * @return A task option to use to ignore the task options parameter
+   */
+  static armonik::api::grpc::v1::TaskOptions get_no_task_options() {
+    armonik::api::grpc::v1::TaskOptions options;
+    options.set_max_retries(INT32_MIN);
```

Comment:
Is that signed ?

Reply:
OK so the min is omething negative. I would not advise a negative number of retries. 1 as default is good enough

Reply:
The idea here is not to have a "proper default". It's to detect whether the user actually put a TaskOption or not.
The rationale is that we don't want to put explicitely a TaskOption in the request (because otherwise it would be merged by core). The reason for not using the default value is that it can then distinguish between someone who put a task option but forgot to set the max_retries value, and someone who doesn't want to put an overriding task option.

Another solution could be a pointer, but I'm not a fan of leaving pointers in pod structures

Reply:
null is not enough ?

Reply:
Iif you have a pointer yes, otherwise you need some sort of mechanism to detect whether the user has put an actual value